### ISSUE TITLE
Feature request: Ability to specify string literals without double quotes in notations

### ISSUE THREAD

**Issue by vmurali** [coq/coq#18659]
For instance, I want to be able to do this:

```
NotationStringArgs "'Foo' literalString" := (Bar "literalString"%string) (at level 20).
(* Foo 25 = Bar "25". *)
(* Foo xkcd = Bar "xkcd". *)
```

**Comment by JasonGross** [comment]
I think the right way to do this would be to be able to specify string notation delimiters when declaring a `String Notation`, and possibly have a restriction that if the entry the string notation is attracted to is not empty and the delimiters are empty, then the string notation only parses idents or something like that.

Thoughts @coq/extensible-syntax-maintainers?

**Comment by vmurali** [comment]
I am trying to avoid any delimiters for strings in these notations. Essentially use a string based lookup but without exposing that to the user and avoid having to specify a new notation for every possible string. The idea is to treat every argument in this notation as a string without delimiters

**Comment by JasonGross** [comment]
Yes, the feature I'm suggesting allows for the possibility of empty delimiters, but is more general than that.  You would write something like
```coq
Declare Scope literal_string_scope.
Delimit Scope literal_string_scope with literal_string.
String Notation string string_of_list_byte list_byte_of_string (delimited by "" "") : literal_string_scope.
Notation "'Foo' literalString" := (Bar literalString%literal_string) (at level 20).
(* Foo 25 = Bar "25". *)
(* Foo xkcd = Bar "xkcd". *)
```
The `delimited by` takes a pair of strings (maybe you could parenthesize them and get a comma separated list of strings), each of which is used as a delimiting token.  So the default is equivalent to `delimited by """" """"`, but you could also do something like `delimited by "“" "”"` to get smart-quote delimited strings.

**Comment by samuelgruetter** [comment]
I described some hacks in https://github.com/coq/coq/issues/15643 that might allow @vmurali to implement the desired feature with Ltac2+custom entries, and might inspire the Coq developers to implement the desired feature for real, so that users stop relying on these hacks :wink:

**Comment by vmurali** [comment]
I think I understand what @JasonGross is suggestion now. And thanks @samuelgruetter, I will take a look at your hack :)
